docs/decisions/ADR_TEMPLATE.md

```diff
@@ -0,0 +1,31 @@
+# Decision record template
+
+<!-- Inspired by: https://github.com/joelparkerhenderson/architecture-decision-record/blob/main/templates/decision-record-template-by-michael-nygard/index.md -->
+<!-- Start by updating "Title", and delete everything before it. -->
+
+# Title
+
+## Status
+
+<!-- What is the status? If it was decided, when did that happen? -->
+
+## tl;dr
+
+<!-- What's the tl;dr of the decision? -->
+
+## Context
+
+<!--
+Feel free to elaborate here. Consider including:
+
+- Why the decision was made
+- Alternatives considered
+-->
+
+## Consequences
+
+<!--
+Discuss the consequences or drawbacks of the decided approach.
+If your answer is "there are none," it's possible you haven't
+fully considered the impact of your decision.
```

Comment:
@charlespwd What do you think about including a comment here suggesting to discuss possible "rollback"/"evolution" approaches?

That would be nice to document solutions for future known problems and also give a hint of how to possibly "undo" the decision or part of it.